src/jit/compiler.h

```diff
@@ -3861,13 +3861,26 @@ public :
     // Convert a BYTE which represents the VM's CorInfoGCtype to the JIT's var_types
     var_types   getJitGCType(BYTE gcType);
 
-    // Get the "primitive" type, if any, that is used to pass or return
-    // values of the given struct type.
-    var_types    argOrReturnTypeForStruct(CORINFO_CLASS_HANDLE clsHnd, bool forReturn);
+    enum structPassingKind { SPK_Unknown,             // Invalid value, never returned
+                             SPK_PrimitiveType,       // The struct is passed/returned using a primitive type
+                             SPK_ByValue,             // The struct is passed/returned by value, usually in multiple registers
+                             SPK_ByValueAsHfa,        // The struct is passed/returned by value as an HFA in multiple registers
+                             SPK_ByReference };       // The struct is passed/returned by reference to a copy/buffer
+
+    // Get the "primitive" type that is is used when we are given a struct of size 'structSize'
+    // when give a pointer sized struct the clsHnd is used to determine if the struct contains GC ref
+    var_types    getPrimitiveTypeForStruct(unsigned structSize, CORINFO_CLASS_HANDLE clsHnd);
```

Comment:
It would be nice to add periods at the end of sentences - I think "when" should start a new sentence but it's not entirely clear.